### ISSUE TITLE
Adding ruff as supported linter in `python.formatting.provider`

### ISSUE THREAD

**Issue by DataHearth** [charliermarsh/ruff-vscode#65]
Hey!

I've been using the extension quite a lot lately, I've been wondering if there's a way to add the ruff linter as an available option inside the `python.formatting.provider` list. 

I can handle it if needed. I think it'd be a geat addition to integrate the linter inside vscode `ms-python` extension. Instead of having `black` as default formatter and ruff doing another work right after ^^.

NB: I don't know if this is the right repo for this ^^

**Comment by charliermarsh** [comment]
I think there was discussion on this at one point, and the Microsoft team said that they'd prefer for others to implement their own extensions rather than bundle more behavior into the Python extension. Let me see if I can dig up the issue... (I may also be misunderstanding the relationship between some of these things.)

**Comment by charliermarsh** [comment]
This is what I was thinking of: https://github.com/microsoft/vscode-python/issues/19843.

**Comment by DataHearth** [comment]
@charliermarsh yeah, I also find that one before researching this topic. So I might not understand how everything works. When hitting the save macro (usually `ctrl+s`), it triggers an event to lint on save and organize imports on save, right ? But, if you have the `ms-python` installed, it'll also trigger the `ms-python` events instead of just the `ruff-vscode` extension. Which is kinda annoying as I highly prefer your extension and the `ruff` tool to black or any other linter ^^.

In this case, should we have an instruction `"editor.defaultFormatter": "charliermarsh.ruff"` under a section `["python"]` in `settings.json` ? I tried this morning, it seems to work. Should I add a small line in the `README.md` to indicate that if someone is willing to keep the `ms-python` extension but use the `ruff-vscode` extension, he/she should use this line ?

**Comment by charliermarsh** [comment]
Yeah, I tested this out, and I think you're right, that this works:

```json
{
    "[python]": {
        "editor.defaultFormatter": "charliermarsh.ruff",
        "editor.formatOnSave": true,
        "editor.codeActionsOnSave": {
            "source.fixAll": true
        },
    }
}
```

I'm confused, because I would've expected _this_ to work:

```json
{
    "[python]": {
        "editor.formatOnSave": true,
        "editor.codeActionsOnSave": {
            "source.fixAll": true
        },
    },
    "python.formatting.provider": "none",
}
```

But that still runs Black.
